gubernator/kubelet_parser.py

```diff
@@ -37,17 +38,30 @@ def parse(lines, error_re, hilight_words, filters):
     """
     matched_lines = []
     uid = ""
+    namespace = ""
 
-    end = 0
     for n, line in enumerate(lines):
         if error_re.search(line):
             matched_lines.append(n)
-            if filters["uid"] and uid == "":
-                s = regex.uidobj_re.search(line)
-                if s and s.group(1) != "":
-                    end = n
-                    uid = s.group(1)
+
+            # If the line is the ObjectReference line, make a dictionary
+            objref = regex.objref(line)
+            if objref and objref.group(1) != "":
+                objref_dict = objref.group(1)        
+                keys = regex.keys_re.findall(objref_dict)
+                
+                for k in keys:
+                    objref_dict = regex.key_to_string(k, objref_dict)
+                
+                # Convert string into dictionary
+                objref_dict = ast.literal_eval(regex.fix_quotes(objref_dict))
+
+                if filters["uid"] and uid == "":
+                    uid = objref_dict["UID"]
                     hilight_words.append(uid)
```

Comment:
I meant check that `objref_dict` has a non-empty `"UID"` entry. It's possible that a partial ObjectRef could be logged in some cases.